### PR TITLE
Update archive job to use COMIN/COMOUT

### PR DESCRIPTION
**Pull request by HenryRWinterbottom**
# Description

NCO has requested that each COM variable specify whether it is an input or an output. This completes that process for the global-workflow archive task.

  Refs #2451 

# Type of change
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?

- `C48_ATM` experiment on RDHPCS Hera
- CI/CD will test the additional applications

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Review by DavidHuber-NOAA — CHANGES_REQUESTED** [review]
The archive job should have read-only (i.e. `COMIN`) access all `COM` directories _except_ `COMOUT_ATMOS_TRACK`, in which a couple of files are written.  Everything else should be `COMIN`.

**Comment by HenryRWinterbottom** [comment]
Thank you for the feedback, @DavidHuber-NOAA. I remain uncertain about the convention here. I'd like to wait on @WalterKolczynski-NOAA's feedback before I address this suggestion. If it is `COMIN` rather than `COMOUT`, it is a trivial fix. But, I just want to be sure that it complies with what NCO needs/wants here.

**Comment by WalterKolczynski-NOAA** [comment]
> Thank you for the feedback, @DavidHuber-NOAA. I remain uncertain about the convention here. I'd like to wait on @WalterKolczynski-NOAA's feedback before I address this suggestion. If it is `COMIN` rather than `COMOUT`, it is a trivial fix. But, I just want to be sure that it complies with what NCO needs/wants here.

If you are reading pre-existing files, it is COMIN.

If you are writing files, it is COMOUT.

If it is doing both, talk to me.

**Comment by HenryRWinterbottom** [comment]
Thank you, @WalterKolczynski-NOAA. So, in this case, since we are only reading the files such that they are being written to the archive, `COMIN` is the attribute required as noted by @DavidHuber-NOAA. Is that correct?

**Comment by DavidHuber-NOAA** [comment]
@WalterKolczynski-NOAA Thanks for the clarification.  There is one `COM` directory that is read and written to and that is `COM_ATMOS_TRACK`.  It probably makes a difference that this job is not run by NCO.  What do you think we should call it?

@HenryWinterbottom-NOAA Besides that, everything should be `COMIN`.

**Comment by WalterKolczynski-NOAA** [comment]
> @WalterKolczynski-NOAA Thanks for the clarification. There is one `COM` directory that is read and written to and that is `COM_ATMOS_TRACK`. It probably makes a difference that this job is not run by NCO. What do you think we should call it?
> 
> @HenryWinterbottom-NOAA Besides that, everything should be `COMIN`.

The track directory is an odd duck all around right now we're going to have to do something with. For now, define both a `COMIN_ATMOS_TRACK` and a `COMOUT_ATMOS_TRACK`.

**Comment by WalterKolczynski-NOAA** [comment]
> Thank you, @WalterKolczynski-NOAA. So, in this case, since we are only reading the files such that they are being written to the archive, `COMIN` is the attribute required as noted by @DavidHuber-NOAA. Is that correct?

Correct

**Review by DavidHuber-NOAA — APPROVED** [review]
Looks good to me!

File: jobs/JGLOBAL_ARCHIVE
Comment (WalterKolczynski-NOAA) [inline jobs/JGLOBAL_ARCHIVE:12]: Please use only spaces to indent. Mixing in tabs is one of the reasons you keep having indentation errors.

File: parm/archive/arcdir.yaml.j2
Comment (WalterKolczynski-NOAA) [inline parm/archive/arcdir.yaml.j2]: These all look like COMIN to me (data is being copied FROM `ATMOS_TRACK`).

File: scripts/exglobal_archive.py
Comment (WalterKolczynski-NOAA) [inline scripts/exglobal_archive.py]:
```suggestion
        if key.startswith("COMIN_") or key.startswith("COMOUT_"):
```

File: ush/python/pygfs/task/archive.py
Comment (WalterKolczynski-NOAA) [inline ush/python/pygfs/task/archive.py]: The lines following this will need to be updated to use the appropriate track_dir (even if they are both the same right now).

```suggestion
        track_dir_in = arch_dict.COMIN_ATMOS_TRACK
        track_dir_out = arch_dict.COMOUT_ATMOS_TRACK
```

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: ush/python/pygfs/task/archive.py
Comment (aerorahul) [inline ush/python/pygfs/task/archive.py:378]: this change needs to be reflected on lines 383, 397. for e.g. Otherwise, those lines will fail.

File: jobs/JGLOBAL_ARCHIVE
Comment (aerorahul) [inline jobs/JGLOBAL_ARCHIVE]: misplaced indent.

**Review by aerorahul — CHANGES_REQUESTED** [review]
The job will fail archiving the tracks.

File: ush/python/pygfs/task/archive.py
Comment (HenryRWinterbottom) [reply ush/python/pygfs/task/archive.py:378]: It is not clear to me what you are asking or referring to. Please clarify.

Comment (WalterKolczynski-NOAA) [reply ush/python/pygfs/task/archive.py:378]: `track_dir` was replaced with `track_dir_in` and `track_dir_out`. All the subsequent references to `track_dir` need to be changed to one of those two.

File: ush/python/pygfs/task/archive.py
Comment (aerorahul) [inline ush/python/pygfs/task/archive.py]: This should be:
```suggestion
            in_track_file = (track_dir_in + "/avno.t" +
```

File: ush/python/pygfs/task/archive.py
Comment (aerorahul) [inline ush/python/pygfs/task/archive.py]: This should be:
```suggestion
            in_track_p_file = (track_dir_in + "/avnop.t" +
```

File: ush/python/pygfs/task/archive.py
Comment (aerorahul) [reply ush/python/pygfs/task/archive.py:378]: changes were made in a later commit.

**Review by aerorahul — APPROVED** [review]
changes look good.
please ensure the archive job succeeds and places the necessary data in their respective target locations.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved

**Comment by aerorahul** [comment]
@HenryWinterbottom-NOAA 
Can you please resolve the conflict in this PR.  It should be relatively easy.  Please let me know if you need assistance.

Also, cannot run the tests on WCOSS2 yet as the libraries have not been synced on dogwood since the production switch yesterday.  There is a WCOSS2 helpdesk ticket open for it.

**Comment by HenryRWinterbottom** [comment]
Done.

File: parm/archive/gfsa.yaml.j2
Comment (aerorahul) [inline parm/archive/gfsa.yaml.j2]: After #2624 was merged, this section was introduced.
`COM_OBS` should become `COMIN_OBS` otherwise this will fail if `DO_PREP_OBS_AERO` is `YES`.  
Currently, this is a `NO`, so there is no test coverage for this yet.

Comment (HenryRWinterbottom) [reply parm/archive/gfsa.yaml.j2]: Done.

**Comment by HenryRWinterbottom** [comment]
@aerorahul The `C48_S2SW` completed without issue related to the recent updates.

File: parm/archive/gdasice.yaml.j2
Comment (DavidHuber-NOAA) [inline parm/archive/gdasice.yaml.j2]: This should still be `FHOUT_ICE`.  `FHOUT_OCNICE` is not defined.

```suggestion
        {% for fhr in range(FHOUT_ICE, FHMAX+1, FHOUT_ICE) %}
```

File: parm/archive/gdasocean.yaml.j2
Comment (DavidHuber-NOAA) [inline parm/archive/gdasocean.yaml.j2]:
```suggestion
        {% for fhr in range(FHMIN, FHMAX + 1, FHOUT_OCN) %}
```

**Review by DavidHuber-NOAA — CHANGES_REQUESTED** [review]
The `C48mx500_3DVarAOWCDA` test case is failing due to undefined variables in the `gdasocean` and `gdasice` jinja templates.  These should be changed back to their original values.

**Comment by DavidHuber-NOAA** [comment]
@HenryWinterbottom-NOAA `jobs/JGDAS_ENKF_ARCHIVE` and `scripts/exgdas_enkf_earc.py` also need to be updated.  I all of the COM variables in the following lines should be updated to `COMIN`:

https://github.com/HenryWinterbottom-NOAA/global-workflow/blob/dacf10ba5d93f3db518f7759c89c4f35989b7d5c/jobs/JGDAS_ENKF_ARCHIVE#L13-L15

https://github.com/HenryWinterbottom-NOAA/global-workflow/blob/dacf10ba5d93f3db518f7759c89c4f35989b7d5c/scripts/exgdas_enkf_earc.py#L39

This is the cause of the C96C48_hybatmDA CI failure.

**Comment by HenryRWinterbottom** [comment]
@aerorahul This is ready for CI/CD again.

Thank you, @DavidHuber-NOAA.

**Comment by DavidHuber-NOAA** [comment]
@HenryWinterbottom-NOAA I think the ENKF scripts still need to be updated.  I couldn't put in suggestions and commit them since they weren't touched previously in your work.

**Comment by HenryRWinterbottom** [comment]
@DavidHuber-NOAA Thank you. I missed those before I sent the last comment. Should be ready now.

File: jobs/JGDAS_ENKF_ARCHIVE
Comment (aerorahul) [inline jobs/JGDAS_ENKF_ARCHIVE]: The templates do not have `COMIN` in their names.

**Comment by aerorahul** [comment]
@HenryWinterbottom-NOAA The templates do not have `COMIN` names.

**Comment by HenryRWinterbottom** [comment]
> @HenryWinterbottom-NOAA The templates do not have `COMIN` names.

Thank you. Fixed.

File: jobs/JGDAS_ENKF_ARCHIVE
Comment (aerorahul) [reply jobs/JGDAS_ENKF_ARCHIVE]: Thank you.

**Review by DavidHuber-NOAA — APPROVED** [review]
Looks good, thanks!

**Comment by DavidHuber-NOAA** [comment]
Starting CI on WCOSS2.

**Comment by DavidHuber-NOAA** [comment]
@TerrenceMcGuinness-NOAA WCOSS2 failed to clone the branch.  Does the controller on Dogwood need to be updated/installed?

**Comment by DavidHuber-NOAA** [comment]
All tests passed on Hera.  Marking as such.

File: parm/archive/master_enkf.yaml.j2
Comment (aerorahul) [inline parm/archive/master_enkf.yaml.j2:67]: These should be `COM_ATMOS_<>_TMPL`. 
This was the cause of failure reported in https://github.com/NOAA-EMC/global-workflow/pull/2667#issuecomment-2176655888

Comment (HenryRWinterbottom) [reply parm/archive/master_enkf.yaml.j2:67]: Fixed.

Comment (aerorahul) [reply parm/archive/master_enkf.yaml.j2:67]: This PR was merged.  The fix needs a hotfix.

File: scripts/exgdas_enkf_earc.py
Comment (aerorahul) [inline scripts/exgdas_enkf_earc.py:39]: Ref. the failure in https://github.com/NOAA-EMC/global-workflow/pull/2667#issuecomment-2177032111.
Following the hotfix in #2697, those `COM_ATMOS_<>_TMPL` variables are also needed to resolve the jinja2 templates (in addition to the `COMIN`)
We need another hotfix for this.

Comment (HenryRWinterbottom) [reply scripts/exgdas_enkf_earc.py:39]: Do you want to open an issue or just for me to submit a hotfix? Or, did you already fix this?

Comment (WalterKolczynski-NOAA) [reply scripts/exgdas_enkf_earc.py:39]: Fixed it as part of #2667

**Pull request merged**
